src/libexpr/paths.cc

```diff
@@ -0,0 +1,96 @@
+#include "eval.hh"
+#include "util.hh"
+#include "fs-input-accessor.hh"
+
+namespace nix {
+
+SourcePath EvalState::rootPath(const Path & path)
+{
+    return {rootFS, CanonPath(path)};
+}
+
+void EvalState::registerAccessor(ref<InputAccessor> accessor)
+{
+    inputAccessors.emplace(accessor->number, accessor);
+}
+
+static constexpr std::string_view marker = "/__virtual__/";
```

Comment:
How about using `__nix_virtual__`? Then it'd be clearer that this is a Nix-related thing.